### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by altro3**
1. Javadoc lint fixes
2. Checkstyle fixes
3. Some typos fixes
4. Fixed some tests for different system locale
5. Fixed minimal java release for inject-test JavaParser
6. Added '--add-opens java.base/java.lang=ALL-UNNAMED' for correct work SystemLambda and '--add-opens jdk.compiler/com.sun.tools.javac.lang=ALL-UNNAMED' for kotlin in tests
7. Fixed DefaultFileSystemResourceLoader normalize method for windows

### CODE REVIEW

File: inject/src/main/java/io/micronaut/inject/ParametrizedProvider.java [hunk 1]
```diff
@@ -1,47 +0,0 @@
...
-import java.util.Map;
-
-/**
- * An extended version of the {@link Provider} interface for {@link ParametrizedBeanFactory}.
- *
- * @param <T> The type
- * @author graemerocher
- * @since 1.0
- */
-public interface ParametrizedProvider<T> extends Provider<T> {
```
Comment (graemerocher) [inline inject/src/main/java/io/micronaut/inject/ParametrizedProvider.java]: this is a breaking change we can't remove it even if it is not used until next major

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: inject/src/main/java/io/micronaut/inject/ParametrizedProvider.java (hunk 1 above)
Comment (altro3) [reply inject/src/main/java/io/micronaut/inject/ParametrizedProvider.java]: @graemerocher And how to be? Will this class be present in all releases up to version 5.0 now?

As far as I understand, you forgot to remove it when removed the `ParametrizedBeanFactory` class.

Comment (altro3) [reply inject/src/main/java/io/micronaut/inject/ParametrizedProvider.java]: Anyway I rolled back this change

**Comment by altro3** [comment]
@sdelamo @dstepanov Hi! Could you review this changes?

File: context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java [hunk 2]
```diff
@@ -45,7 +45,7 @@ public enum ExecutorType {
     WORK_STEALING,
 
     /**
-     * @see java.util.concurrent.Executors#newThreadPerTaskExecutor()
+     * @see io.micronaut.scheduling.LoomSupport#newThreadPerTaskExecutor(java.util.concurrent.ThreadFactory)
```
Comment (sdelamo) [inline context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java]: @yawkat is this change in the javadoc correct?

File: core/src/main/java/io/micronaut/core/io/file/DefaultFileSystemResourceLoader.java [hunk 3]
```diff
@@ -129,6 +129,9 @@ private static String normalize(String path) {
         }
         if (path.startsWith("file:")) {
             path = path.substring(5);
+            if (path.startsWith("//")) {
```
Comment (sdelamo) [inline core/src/main/java/io/micronaut/core/io/file/DefaultFileSystemResourceLoader.java:132]: Why this change? can you add a test?

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/body/MultiObjectBody.java [hunk 4]
```diff
@@ -36,7 +36,7 @@
 public sealed interface MultiObjectBody extends HttpBody permits ImmediateMultiObjectBody, ImmediateSingleObjectBody, StreamingMultiObjectBody {
     /**
      * Coerce this value to an {@link InputStream}. This implements
-     * {@link io.micronaut.http.server.netty.binders.InputStreamBodyBinder}. Requires the objects
+     * io.micronaut.http.server.netty.binders.NettyInputStreamBodyBinder. Requires the objects
```
Comment (sdelamo) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/body/MultiObjectBody.java]: why remove the `@link`?

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java [hunk 5]
```diff
@@ -1071,7 +1071,7 @@ public static class Parent extends EventLoopConfig {
     }
 
     /**
-     * Allows configuration of properties for the {@link io.micronaut.http.server.netty.types.files.FileTypeHandler}.
+     * Allows configuration of properties for the {@link io.micronaut.http.server.netty.body.AbstractFileBodyWriter}.
```
Comment (sdelamo) [inline http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java]: @yawkat is this change of javadoc correct?

File: inject-kotlin-test/src/main/kotlin/io/micronaut/annotation/processing/test/support/AbstractKotlinCompilation.kt [hunk 6]
```diff
@@ -293,7 +293,6 @@ abstract class AbstractKotlinCompilation<A : CommonCompilerArguments> internal c
 
 internal fun convertKotlinExitCode(code: ExitCode) = when(code) {
     ExitCode.OK -> KotlinCompilation.ExitCode.OK
-    ExitCode.OOM_ERROR,
```
Comment (sdelamo) [inline inject-kotlin-test/src/main/kotlin/io/micronaut/annotation/processing/test/support/AbstractKotlinCompilation.kt]: why this removal?

File: inject/build.gradle [hunk 7]
```diff
@@ -30,8 +30,11 @@ dependencies {
 
 tasks.withType(Test).configureEach {
     if (JavaVersion.current().majorVersion.toInteger() >= 17) {
-        logger.warn("Opening java.util, so SystemLambda can work")
-        jvmArgs += ['--add-opens', 'java.base/java.util=ALL-UNNAMED']
+        logger.warn("Opening java.util and java.lang, so SystemLambda can work")
+        jvmArgs += [
+                '--add-opens', 'java.base/java.util=ALL-UNNAMED',
+                '--add-opens', 'java.base/java.lang=ALL-UNNAMED',
```
Comment (sdelamo) [inline inject/build.gradle]: why is this necessary?

File: jackson-core/src/main/java/io/micronaut/jackson/core/tree/TreeGenerator.java [hunk 8]
```diff
@@ -167,7 +167,7 @@ public JsonNode getCompletedValue() {
     }
 
     @Override
-    public void writeStartArray() throws IOException {
+    public void writeStartArray() {
```
Comment (sdelamo) [inline jackson-core/src/main/java/io/micronaut/jackson/core/tree/TreeGenerator.java:170]: why removing the `throws`. Does not throw IOException anymore?

File: jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java [hunk 9]
```diff
@@ -104,6 +97,13 @@ private JacksonDatabindMapper(JacksonDatabindMapper from, Argument<?> type) {
         this.specializedWriter = from.createWriter(type);
     }
 
+    private static ObjectMapper createDefaultMapper() {
```
Comment (sdelamo) [inline jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java]: why this change? does it just move the method or I am missing something?

File: runtime/build.gradle [hunk 10]
```diff
@@ -50,7 +50,10 @@ spotless {
 
 tasks.withType(Test).configureEach {
     if (JavaVersion.current().majorVersion.toInteger() >= 17) {
-        logger.warn("Opening java.util, so SystemLambda can work")
-        jvmArgs += ['--add-opens', 'java.base/java.util=ALL-UNNAMED']
+        logger.warn("Opening java.util and java.lang, so SystemLambda can work")
+        jvmArgs += [
+                '--add-opens', 'java.base/java.util=ALL-UNNAMED',
+                '--add-opens', 'java.base/java.lang=ALL-UNNAMED',
```
Comment (sdelamo) [inline runtime/build.gradle]: why is this needed ?

File: core/src/main/java/io/micronaut/core/io/file/DefaultFileSystemResourceLoader.java (hunk 3 above)
Comment (altro3) [reply core/src/main/java/io/micronaut/core/io/file/DefaultFileSystemResourceLoader.java:132]: Problem with pathes in windows.... Path like this `//D:/Wok/....` is wrong

File: inject-kotlin-test/src/main/kotlin/io/micronaut/annotation/processing/test/support/AbstractKotlinCompilation.kt (hunk 6 above)
Comment (altro3) [reply inject-kotlin-test/src/main/kotlin/io/micronaut/annotation/processing/test/support/AbstractKotlinCompilation.kt]: duplicate OOM_ERROR:
изображение

File: inject/build.gradle (hunk 7 above)
Comment (altro3) [reply inject/build.gradle]: without it tests doesn't work on Windows and I see exception with SystemLambda

File: jackson-core/src/main/java/io/micronaut/jackson/core/tree/TreeGenerator.java (hunk 8 above)
Comment (altro3) [reply jackson-core/src/main/java/io/micronaut/jackson/core/tree/TreeGenerator.java:170]: You don't need to write exceptions from interface if your implementation don't throw them

File: jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java (hunk 9 above)
Comment (altro3) [reply jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java]: checkstyle order error

File: context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java (hunk 2 above)
Comment (yawkat) [reply context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java]: no. the old one is correct. i assume it triggers a lint failure because it points to a method that is not available on java 17, but that's fine imo.

File: runtime/build.gradle (hunk 10 above)
Comment (altro3) [reply runtime/build.gradle]: without it tests doesn't work on Windows and I see exception with SystemLambda

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/body/MultiObjectBody.java (hunk 4 above)
Comment (altro3) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/body/MultiObjectBody.java]: Hm... this is maybe a byg in IDEA. I see now this picture:
изображение
But I also don't understand what is wrong...

Reverted

File: context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java (hunk 2 above)
Comment (altro3) [reply context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java]: @yawkat hm... yes, it's true. But why it's Ok if we use java 17 and java 17 doesn't have this method in JDK ?

Comment (yawkat) [reply context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java]: this setting does not work on java 17 anyway. only on jdks where this method is available does it work.

Comment (altro3) [reply context/src/main/java/io/micronaut/scheduling/executor/ExecutorType.java]: I wonder if there are such JDKs? ok i will revert this change

File: http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java (hunk 5 above)
Comment (altro3) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java]: @yawkat Hi! Could you answer? I think, this is last unanswered question

Comment (yawkat) [reply http-server-netty/src/main/java/io/micronaut/http/server/netty/configuration/NettyHttpServerConfiguration.java]: yes

File: aop/src/main/kotlin/io/micronaut/aop/util/KotlinInterceptedMethodHelper.kt [hunk 11]
```diff
@@ -35,8 +35,8 @@ internal object KotlinInterceptedMethodHelper {
     suspend fun handleResult(result: CompletionStage<*>, isUnitValueType: Boolean): Any? = suspendCoroutine { continuation ->
         result.whenComplete { value: Any?, throwable: Throwable? ->
             if (throwable == null) {
-                val result = Result.success(value ?: if (isUnitValueType) Unit else null)
-                continuation.resumeWith(result)
+                val res = Result.success(value ?: if (isUnitValueType) Unit else null)
```
Comment (chalkedgoose) [inline aop/src/main/kotlin/io/micronaut/aop/util/KotlinInterceptedMethodHelper.kt:38]: This change seems a bit superfluous is there a reason for renaming the symbol from 
result to res? I mean you could make a case for a more explicit name but in this case 
res does not feel like an improvement over result.

edit: if these changes are related to linting or checkstyle in any way just ignore this 
comment and the other I made below.

File: test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java [hunk 12]
```diff
@@ -37,7 +37,7 @@
 @SupportedSourceVersion(RELEASE_17)
 public class TestGeneratingAnnotationProcessor extends AbstractProcessor {
 
-    private boolean executed = false;
+    private boolean executed;
```
Comment (chalkedgoose) [inline test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java:40]: There is no concrete difference between these two lines I don't want 
to be overly nit picky but what are the reasons for this change?

File: aop/src/main/kotlin/io/micronaut/aop/util/KotlinInterceptedMethodHelper.kt (hunk 11 above)
Comment (altro3) [reply aop/src/main/kotlin/io/micronaut/aop/util/KotlinInterceptedMethodHelper.kt:38]: @chalkedgoose The problem is that you use two variables with the same name - a method parameter and a local variable in the lambda. This makes your code less clear. It's not obvious at what point you're using the parameter, but at what variable. Idea inspection highlighted me this problem, I didn't see it myself. Although, it's strange why sonar didn't catch the smell code before I did

File: test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java (hunk 12 above)
Comment (altro3) [reply test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java:40]: @chalkedgoose It makes no sense to assign default values to primitives. You don't write an assignment next to every attribute of the class, so why should you assign a default value here? Anyway, it's just a smell code correction.

Comment (yawkat) [reply test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java:40]: i disagree with the linter, i think in this case `= false` is more clear, it shows the value is initially false and not e.g. set in a constructor. Though it's not a big deal.

Comment (altro3) [reply test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java:40]: @yawkat Well, then we need to decide: either we rely on the knowledge of java, or we are against abbreviations and write always default values, so that the code is "cleaner and understandable". That is, you need to correct in the other direction, classes, like:
```java
class MyClass {

    private String f1;
    private float f2;
    private Double f3;
    private int f4;
...
}
```
It should be described as follows:
```java
class MyClass {

    private String f1 = null;
    private float f2 = 0F;
    private Double f3 = null;
    private int f4 = 0;
...
}
```
It's getting really weird right now. Everyone knows that the default boolean value is false, and in most cases the default value is not written, and it is written in this place. So what's the difference between this place and the others?

Comment (yawkat) [reply test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java:40]: the difference is entirely subjective. in some places it's more clear to have an explicit initializer, in others it is not.

Comment (altro3) [reply test-suite-helper/src/main/java/io/micronaut/testsuitehelper/TestGeneratingAnnotationProcessor.java:40]: I will just say that everyone decides for themselves what is more convenient for them. I still attribute this change to the code style. If you think this is not a necessary change, no problem, I don't insist ;-)

File: inject/build.gradle (hunk 7 above)
Comment (sdelamo) [reply inject/build.gradle]: merged in https://github.com/micronaut-projects/micronaut-core/pull/9807

File: inject-kotlin-test/src/main/kotlin/io/micronaut/annotation/processing/test/support/AbstractKotlinCompilation.kt (hunk 6 above)
Comment (sdelamo) [reply inject-kotlin-test/src/main/kotlin/io/micronaut/annotation/processing/test/support/AbstractKotlinCompilation.kt]: Merged in https://github.com/micronaut-projects/micronaut-core/pull/9854

File: jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java (hunk 9 above)
Comment (sdelamo) [reply jackson-databind/src/main/java/io/micronaut/jackson/databind/JacksonDatabindMapper.java]: resolved via https://github.com/micronaut-projects/micronaut-core/pull/9856

File: runtime/build.gradle (hunk 10 above)
Comment (sdelamo) [reply runtime/build.gradle]: resolved via https://github.com/micronaut-projects/micronaut-core/pull/9807/files

File: aop/src/main/kotlin/io/micronaut/aop/util/KotlinInterceptedMethodHelper.kt (hunk 11 above)
Comment (sdelamo) [reply aop/src/main/kotlin/io/micronaut/aop/util/KotlinInterceptedMethodHelper.kt:38]: I think it is a good change. Having different variable names is more clear.

**Comment by sdelamo** [comment]
Closing this as I think we have incorporated almost every suggestion. @altro3 please reopen separate prs if you think we need them.
